## Teste a hipótese de a renda ser igual para os dois estados civis. Qual sua conclusão?

In [1]:
# importa biblioteca pandas
import pandas as pd

# importa biblioteca scipy
import scipy.stats as stats


In [2]:
# local do arquivo
Caminho = '/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'


In [3]:
#Traz o arquivo para o Python
df=pd.read_excel(Caminho)


In [4]:
df.head( )

ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out

In [5]:
#Verifica a média de idade por grupo
df.groupby(['estado_civil'])['idade'].describe()


count       mean       std   min   25%   50%   75%   max
estado_civil                                                          
c             244.0  35.094262  5.761670  25.0  31.0  35.0  39.0  49.0
s             182.0  33.989011  7.130968  20.0  27.0  34.0  41.0  46.0

In [6]:
solteiro=df[df.estado_civil=='s'].idade
casado=df[df.estado_civil=='c'].idade


In [7]:
# Executa o teste de Levene para verificar a igualdade das variâncias
stats.levene(casado, solteiro)

LeveneResult(statistic=15.882848929731193, pvalue=7.932664725672226e-05)

In [8]:
# executa o teste t Testando H0: μc= μs   H1: μc ≠ μs
stats.ttest_ind(casado,solteiro,equal_var=False)
# PARA 95% DE CONFIANÇA REJEITO H0 SE PVALUE < 0.05

TtestResult(statistic=1.7147489742062125, pvalue=0.08730214339336337, df=340.12128870439)

In [9]:
#pvalue = 0,087 maoir que 0,05 aceito a hipotese de igualdade

In [10]:
df.groupby(['estado_civil'])['salario'].describe()

count       mean       std  min   25%    50%    75%   max
estado_civil                                                           
c             244.0  11.854303  4.340324  4.0  8.49  11.14  14.71  23.9
s             182.0  10.043462  3.815224  4.0  7.24   9.20  11.79  18.9

In [11]:
solteiro=df[df.estado_civil=='s'].salario
casado=df[df.estado_civil=='c'].salario


In [12]:
# executa o teste t Testando H0: μc= μs   H1: μc ≠ μs
stats.ttest_ind(casado,solteiro,equal_var=False)
# PARA 95% DE CONFIANÇA REJEITO H0 SE PVALUE < 0.05
#rejeitado

TtestResult(statistic=4.567472731259726, pvalue=6.527014259249644e-06, df=412.678015786978)

### Questao 4

#### Para testar a hipótese de que a renda é igual para os dois estados civis (solteiro e casado), podemos usar o teste t para duas amostras independentes. Vamos seguir os seguintes passos:
### Visualização

#### 1. Extrair os dados de renda para solteiros e casados.
#### 2. Realizar o teste t para duas amostras independentes.
#### 3. Interpretar o valor p para aceitar ou rejeitar a hipótese nula.

### Passos em pseudocódigo:

1. **Importar bibliotecas necessárias** (pandas e scipy.stats).
2. **Carregar os dados do arquivo Excel**.
3. **Extrair as colunas de renda para solteiros e casados**.
4. **Realizar o teste t para duas amostras independentes**.
5. **Interpretar o resultado com base no valor p**.



In [13]:
# Instalar scipy
%pip install scipy pandas

Note: you may need to restart the kernel to use updated packages.


In [14]:
import pandas as pd
from scipy import stats

In [15]:
# Carregar os dados do arquivo Excel
Caminho = '/Users/fabicampanari/Desktop/_8-Prova Matematematica/1-statiscalMeasures_ Hypothesis Testing II./1🇧🇷-statiscalMeasures_ Hypothesis Testing II./answeredCodes_statiscalMeasures/cadastro_funcionarios.xlsx'
df = pd.read_excel(Caminho)

In [16]:
# Visualizar as primeiras linhas do DataFrame
print(df.head())

   ID estado_civil grau_instrucao  n_filhos  salario  idade reg_proc
0   1            s              f         0     4.00     26      int
1   2            c              f         1     4.56     32      cap
2   3            c              f         2     5.25     36      cap
3   4            s              m         0     5.73     20      out
4   5            s              f         0     6.26     40      out


In [17]:
# Verificar a média de salário por grupo de estado civil
print(df.groupby(['estado_civil'])['salario'].describe())

              count       mean       std  min   25%    50%    75%   max
estado_civil                                                           
c             244.0  11.854303  4.340324  4.0  8.49  11.14  14.71  23.9
s             182.0  10.043462  3.815224  4.0  7.24   9.20  11.79  18.9


In [18]:
# Extrair as colunas de renda para solteiros e casados
solteiro = df[df['estado_civil'] == 's']['salario']
casado = df[df['estado_civil'] == 'c']['salario']
casado


1       4.56
2       5.25
5       6.66
8       7.59
10      8.12
       ...  
420    17.80
422    19.30
423    23.50
424    23.90
425    21.40
Name: salario, Length: 244, dtype: float64

In [19]:
# Realizar o teste t para duas amostras independentes
t_stat, p_value = stats.ttest_ind(casado, solteiro, equal_var=False)

In [20]:
# Exibir os resultados do teste t
print("Resultados do Teste t:")
print(f"Estatística t: {t_stat}")
print(f"Valor p: {p_value}")

Resultados do Teste t:
Estatística t: 4.567472731259726
Valor p: 6.527014259249644e-06


In [21]:
# Interpretar o resultado
alpha = 0.05
if p_value < alpha:
    print("Conclusão: Rejeitamos a hipótese nula. As rendas são diferentes para os dois estados civis.")
else:
    print("Conclusão: Não rejeitamos a hipótese nula. As rendas são iguais para os dois estados civis.")

Conclusão: Rejeitamos a hipótese nula. As rendas são diferentes para os dois estados civis.


### Conclusão Resultados do Teste t:

#### Interpretação:
 - Estatística t: 4.567472731259726 <br>
 - Valor p: 6.527014259249644e-06


O valor p é extremamente pequeno (6.527014259249644e-06), muito menor que o nível de significância comum (0.05). Isso indica que há uma diferença significativa nas rendas entre os dois estados civis.

Conclusão:

Rejeitamos a hipótese nula. As rendas são diferentes para os dois estados civis (solteiro e casado).


